Terrain layer spawner documentation. #3228 (#860)
o3de/o3de.org: content/docs/user-guide/gems/reference/terrain/layer_spawner.md

> ---

Author: I assume you mean reference/environment/terrain, as in the previous comment - I've moved it there.

title: Terrain Layer Spawner component
linktitle: LayerSpawner
description: ' Open 3D Engine (O3DE) Terrain Layer Spawner reference. '
weight: 100
---

The **Terrain Layer Spawner** component spawns a terrain layer within given bounds, and enables the ordering of multiple layers.  

> The priority is controlled firstly by assigning a layer, either **Foregound**(Highest priority) or **Background**, then by using the **Priority** setting, with a higher number being a higher priority.  

Reviewer: ```suggestion
The priority is controlled by first assigning a layer, either **Foreground**(Highest priority) or **Background**, and then by using the **Priority** setting, with a higher number being a higher priority.  
```

Author: Changed.

> The dimensions of the layer can be configured by modifying the [Box component](/docs/user-guide/components/reference/shape/box-shape) on the same entity.

Reviewer: ```suggestion
You can configure the dimensions of the layer by modifying the [Box component](/docs/user-guide/components/reference/shape/box-shape) on the same entity.
```

Author: Changed.

The **Box component** is required for the **Layer Spawner** to operate.

## Provider ##

[Terrain Gem](/docs/user-guide/gems/reference/terrain)

## Properties ##

| Property | Description | Values | Default |
|-|-|-|-|

> | **Layer Priority** | The priority of the spawner, Foreground has a higher priority than Background. | Foreground or Background | `Foreground` |

Reviewer: ```suggestion
| **Layer Priority** | The priority of the spawner, Foreground, has a higher priority than Background. | Foreground or Background | `Foreground` |
```
Make sure that the punctuation addition clarifies the intended meaning.

Reviewer 2: > 
> 
> Make sure that the punctuation addition clarifies the intended meaning.

I think this should be two sentences based on context from the interface.

```suggestion
| **Layer Priority** | The priority of the spawner. Foreground has a higher priority than Background. | Foreground or Background | `Foreground` |
```

Reviewer: Oh I see now -- yea I read that wrong!

Author: Yes, the two sentence form is better, a colon would have worked as well. Changed.

| **Sub Priority** | Sets the priority of this spawner within the layer. Higher numbers will override lower. | 0 - 10000 | 0 |

> | **Use Ground Plane** | Enable this to provide a default ground plane where no terrain is defined. | Boolean | `True` |

Reviewer: ```suggestion
| **Use Ground Plane** | Enable this setting to provide a default ground plane where no terrain is defined. | Boolean | `True` |
```

Author: Changed.

## TerrainSpawnerRequestBus ##

Use the following request functions with the `TerrainSpawnerRequestBus` EBus interface to communicate with other components of your game.

### GetPriority 

Retrieves the terrain layer priority.

**Parameters**  
Layer \[out\] - Index of the Layer Priority setting (Foreground is 0).
Type: Unsigned Int  

> Priority \[out\] - Value of the sub priority setting. 

Reviewer: ```suggestion
Priority \[out\] - Value of the Sub Priority setting. 
```

Author: Changed

---
title: Layer Spawner component
linktitle: LayerSpawner
description: ' Open 3D Engine (O3DE) Terrain Layer Spawner reference. '
weight: 100
---

{{< preview-migrated >}}

> 

Author: Thanks

> The **Layer Spawner** component defines the bounds of a terrain layer, and enables the ordering of multiple layers.  

Reviewer 3: Same as above, should mention that it actually spawns the region, it doesn't just define the bounds.

> The priority is controlled firstly by assigning a layer, either **Foregound** or **Background**, then by using the **Priority** setting.  

Reviewer 3: We should probably mention that Foreground is a higher priority than Background, and for Priority, we should mention that higher values represent a higher priority within that layer.

Author: This is stated in the Properties section, but I'll add it here as well.

The dimensions of the layer can be configured by modifying the [Box component](/docs/user-guide/components/reference/shape/box-shape) on the same entity.
The **Box component** is required for the **Layer Spawner** to operate.

## Provider ##

[Terrain Gem](/docs/user-guide/gems/reference/terrain)

## Properties ##

| Property | Description | Values | Default |
|-|-|-|-|
| **Layer Priority** | The priority of the spawner, Foreground has a higher priority than Background. | Foreground or Background | `Foreground` |
| **Sub Priority** | Sets the priority of this spawner within the layer. Higher numbers will override lower. | 0 - 10000 | 0 |
| **Use Ground Plane** | Enable this to provide a default ground plane where no terrain is defined. | Boolean | `Disabled` |

## TerrainSpawnerRequestBus ##

Use the following request functions with the `TerrainSpawnerRequestBus` EBus interface to communicate with other components of your game.

### GetPriority 

> Retrieves the entity's local and world transforms.

Reviewer 3: This should be "Retrieves the terrain layer priority".
